src/conn/conn_sweep.c

```diff
@@ -359,8 +366,14 @@ __wt_sweep_create(WT_SESSION_IMPL *session)
 	/*
 	 * Handle sweep does enough I/O it may be called upon to perform slow
 	 * operations for the block manager.
+	 *
+	 * The sweep thread sweeps the lookaside table for outdated records,
+	 * it gets its own cursor for that purpose.
+	 *
+	 * Don't tap the sweep thread for eviction.
 	 */
-	F_SET(session, WT_SESSION_CAN_WAIT);
+	F_SET(session, WT_SESSION_CAN_WAIT |
+	    WT_SESSION_LOOKASIDE_CURSOR | WT_SESSION_NO_EVICTION);
```

Comment:
`evict_lru.c` manages the LRU queue, which application threads can use if they are stuck when the cache is full.  I don't see them as separate things, and I don't think any thread should be "repurposed from whatever it's doing" -- threads only get involved in eviction if they would otherwise be blocked by a full cache (at least with high probability).

Regardless, we're spending a lot of time discussing the naming of one flag.  I've been through and reviewed the use of `WT_SESSION_NO_EVICTION` again, including the pre-existing cases where we had `WT_SESSION_NO_CACHE_CHECK` plus the new cases to avoid re-entering eviction in places that could cause deadlocks.  I don't think changing the name will help at this point since there isn't an obviously better candidate, and `WT_SESSION_NO_EVICTION` matches the existing `WT_BTREE_NO_EVICTION` naming.

Let's leave this one alone.